v3/src/components/tool-shelf/tool-shelf.tsx

```diff
@@ -20,64 +26,96 @@ export const ToolShelf = () => {
         status: "success"
       })
     },
-    toast = useToast(),
-    tableHandler = () => notify('table'),
-    graphHandler = () => notify('graph'),
-    mapHandler = () => notify('map'),
-    sliderHandler = () => notify('slider'),
-    calcHandler = () => notify('calculator'),
-    textHandler = () => notify('text'),
-    pluginsHandler = () => notify('plugins')
+    toast = useToast()
+
+  const row = content?.getRowByIndex(0) as IFreeTileRow
```

Comment:
I forgot to mention, but we should at least nominally continue to support the mosaic layout, so rather than casting here we should probably do a dynamic type test.